lib/src/resources/_resource.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:meta/meta.dart';
+
+import '../../messages.dart';
+import '../client.dart';
+
+abstract class Resource<T extends Message> {
+  @protected
+  final Client _client;
+
+  Resource(this._client);
+
+  @protected
+  @visibleForOverriding
+  String makeUrl(String path) {
```

Comment:
I’m curious, why did you decide to expose this?

Reply:
I suppose this isn't really needed at all anymore. Originally, I created this as the default functionality, and I also created a mixin `ExpansionSupport` that would override this method to support `?expand[0]=foo` values in retrieve requests. That worked great, until I realized that deserializing the responses from these requests would be much more involved than I was prepared for, so I dropped that commit. I could add it back in if you like, it would support making those requests, but would break the `fromJson()` efforts if the expansions were ever used.